packages/taiko-client/driver/soft_blocks/api.go

```diff
@@ -0,0 +1,304 @@
+			s.rpc.L2.ChainID,
+			new(big.Int).SetUint64(reqBody.TransactionBatch.BlockID),
+			reqBody.TransactionBatch.TransactionsList,
+			true,
+		),
+		reqBody.TransactionBatch.BatchMarker,
+		reqBody.TransactionBatch.BlockParams,
+	)
+	if err != nil {
+		return s.returnError(c, http.StatusInternalServerError, err)
+	}
+
+	return c.JSON(http.StatusOK, BuildSoftBlockResponseBody{BlockHeader: header})
+}
+
+// RemoveSoftBlocksRequestBody represents a request body when resetting the backend
+// L2 execution engine soft head.
+type RemoveSoftBlocksRequestBody struct {
+	// @param newLastBlockID uint64 New last block ID of the blockchain, it should
+	// @param not smaller than the canonical chain's highest block ID.
+	NewLastBlockID uint64 `json:"newLastBlockId"`
+}
+
+// RemoveSoftBlocksResponseBody represents a response body when resetting the backend
+// L2 execution engine soft head.
+type RemoveSoftBlocksResponseBody struct {
+	// @param lastBlockID uint64 Current highest block ID of the blockchain (including soft blocks)
+	LastBlockID uint64 `json:"lastBlockId"`
+	// @param lastProposedBlockID uint64 Highest block ID of the cnonical chain
+	LastProposedBlockID uint64 `json:"lastProposedBlockID"`
+	// @param headsRemoved uint64 Number of soft heads removed
+	HeadsRemoved uint64 `json:"headsRemoved"`
+}
+
+// RemoveSoftBlocks removes the backend L2 execution engine soft head.
+//
+//		@Description	Remove all soft blocks from the blockchain beyond the specified block height,
+//	  	@Description	ensuring the latest block ID does not exceed the given height. This method will fail if
+//	  	@Description	the block with an ID one greater than the specified height is not a soft block. If the
+//	  	@Description	specified block height is greater than the latest soft block ID, the method will succeed
+//	  	@Description	without modifying the blockchain.
+//		@Param      	body RemoveSoftBlocksRequestBody true "soft blocks removing request body"
+//		@Accept			json
+//		@Produce		json
+//		@Success		200	{object} RemoveSoftBlocksResponseBody
+//		@Router			/softBlocks [delete]
+func (s *SoftBlockAPIServer) RemoveSoftBlocks(c echo.Context) error {
+	// Parse the request body.
+	reqBody := new(RemoveSoftBlocksRequestBody)
+	if err := c.Bind(reqBody); err != nil {
+		return s.returnError(c, http.StatusUnprocessableEntity, err)
+	}
+
+	// Request body validation.
+	canonicalHeadL1Origin, err := s.rpc.L2.HeadL1Origin(c.Request().Context())
+	if err != nil {
+		return s.returnError(c, http.StatusInternalServerError, err)
+	}
+
+	currentHead, err := s.rpc.L2.HeaderByNumber(c.Request().Context(), nil)
+	if err != nil {
+		return s.returnError(c, http.StatusInternalServerError, err)
+	}
+
+	log.Info(
+		"New soft block removing request",
+		"newLastBlockId", reqBody.NewLastBlockID,
+		"canonicalHead", canonicalHeadL1Origin.BlockID.Uint64(),
+		"currentHead", currentHead.Number.Uint64(),
+	)
+
+	if reqBody.NewLastBlockID < canonicalHeadL1Origin.BlockID.Uint64() {
```

Comment:
If `reqBody.NewLastBlockID  == canonicalHeadL1Origin.BlockID` then `SafeBlock` will be removed, is it expected？